src/bsoncxx/document/value.hpp

```diff
@@ -76,6 +87,20 @@ class BSONCXX_API value {
     value(value&&) noexcept = default;
     value& operator=(value&&) noexcept = default;
 
+    // Serializer functions
+    template <typename T,
+              typename std::enable_if<!std::is_same<T, typename array::view>::value, int>::type = 0>
```

Comment:
An alternative way to do this would be:
```
template <typename T>
explicit value(const T& t) : value({}) {
        // do stuff
}

template <typename T>
explicit value(
        typename std::enable_if<std::is_same<T, array::view>::value, array::view>::type t)
        : value((document::view)t) {}
```

Reply:
Oh nice use of SFINAE! Either approach seems fine to me.